test/extended/deployments/test_deployments.go

```diff
@@ -0,0 +1,101 @@
+package builds
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("deployments: parallel: test deployment", func() {
+	defer g.GinkgoRecover()
+	var (
+		deploymentFixture = exutil.FixturePath("..", "extended", "fixtures", "test-deployment-test.yaml")
+		oc                = exutil.NewCLI("cli-deployment", exutil.KubeConfigPath())
+	)
+
+	g.Describe("test deployment", func() {
+		g.It("should run a deployment to completion and then scale to zero", func() {
+			out, err := oc.Run("create").Args("-f", deploymentFixture).Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			out, err = oc.Run("logs").Args("-f", "dc/deployment-test").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By(fmt.Sprintf("checking the logs for substrings\n%s", out))
+			o.Expect(out).To(o.ContainSubstring("deployment-test-1 to 2"))
+			o.Expect(out).To(o.ContainSubstring("Pre hook finished"))
+			o.Expect(out).To(o.ContainSubstring("Deployment deployment-test-1 successfully made active"))
+
+			g.By("verifying the deployment is marked complete and scaled to zero")
+			err = wait.Poll(100*time.Millisecond, 10*time.Second, func() (bool, error) {
```

Comment:
Makes the code harder to read in this case - however I will increase the poll window